src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts

```diff
@@ -91,13 +91,21 @@ export class AttachmentUnitFormComponent implements OnInit, OnChanges {
         });
     }
 
-    onFileChange(event: any): void {
-        if (event.target.files.length) {
-            const fileList = event.target.files;
-            this.file = fileList[0];
-            this.fileName = this.file.name;
-            this.isFileTooBig = this.file.size > MAX_FILE_SIZE;
+    onFileChange(event: Event): void {
+        const input = event.target as HTMLInputElement;
+        if (!input.files?.length) {
+            return;
+        }
+        this.file = input.files[0];
+        this.fileName = this.file.name;
+        // automatically set the name in case it is not yet specified
+        if (this.form && (this.nameControl?.value == undefined || this.nameControl?.value == '')) {
+            this.form.patchValue({
+                // without extension
+                name: this.file.name.replace(/\.[^/.]+$/, ''),
```

Comment:
Just a suggestion: we could also replace `_` by spaces
```suggestion
                name: this.file.name.replace(/\.[^/.]+$/, '').replace(/_/g, ' ').trim(),
```